### PR TITLE
fix embedding example

### LINKED ISSUE #5909: embedding example is broken
Can't build embeddine example with following output:
```
workstation@workstation:~/dev/lego/micropython$ make -C examples/embedding
make: Entering directory '/home/workstation/dev/lego/micropython/examples/embedding'
cc -std=c99 -I. -I../.. -DNO_QSTR   -c -o hello-embed.o hello-embed.c
In file included from ../../py/reader.h:29:0,
                 from ../../py/lexer.h:33,
                 from ../../py/compile.h:29,
                 from hello-embed.c:31:
../../py/obj.h:733:66: error: unknown type name ‘mp_rom_error_text_t’; did you mean ‘mp_rom_map_elem_t’?
 mp_obj_t mp_obj_new_exception_msg(const mp_obj_type_t *exc_type, mp_rom_error_text_t msg);
                                                                  ^~~~~~~~~~~~~~~~~~~
                                                                  mp_rom_map_elem_t
../../py/obj.h:734:71: error: unknown type name ‘mp_rom_error_text_t’; did you mean ‘mp_rom_map_elem_t’?
 mp_obj_t mp_obj_new_exception_msg_varg(const mp_obj_type_t *exc_type, mp_rom_error_text_t fmt, ...); // counts args by number of % symbols in fmt, excluding %%; can only handle void* sizes (ie no float/double!)
                                                                       ^~~~~~~~~~~~~~~~~~~
                                                                       mp_rom_map_elem_t
In file included from ../../py/bc.h:30:0,
                 from ../../py/emitglue.h:30,
                 from ../../py/compile.h:31,
                 from hello-embed.c:31:
../../py/runtime.h:165:59: error: unknown type name ‘mp_rom_error_text_t’; did you mean ‘mp_rom_obj_tuple_t’?
 NORETURN void mp_raise_msg(const mp_obj_type_t *exc_type, mp_rom_error_text_t msg);
                                                           ^~~~~~~~~~~~~~~~~~~
                                                           mp_rom_obj_tuple_t
../../py/runtime.h:166:64: error: unknown type name ‘mp_rom_error_text_t’; did you mean ‘mp_rom_obj_tuple_t’?
 NORETURN void mp_raise_msg_varg(const mp_obj_type_t *exc_type, mp_rom_error_text_t fmt, ...);
                                                                ^~~~~~~~~~~~~~~~~~~
                                                                mp_rom_obj_tuple_t
../../py/runtime.h:167:35: error: unknown type name ‘mp_rom_error_text_t’; did you mean ‘mp_rom_obj_tuple_t’?
 NORETURN void mp_raise_ValueError(mp_rom_error_text_t msg);
                                   ^~~~~~~~~~~~~~~~~~~
                                   mp_rom_obj_tuple_t
../../py/runtime.h:168:34: error: unknown type name ‘mp_rom_error_text_t’; did you mean ‘mp_rom_obj_tuple_t’?
 NORETURN void mp_raise_TypeError(mp_rom_error_text_t msg);
                                  ^~~~~~~~~~~~~~~~~~~
                                  mp_rom_obj_tuple_t
../../py/runtime.h:169:44: error: unknown type name ‘mp_rom_error_text_t’; did you mean ‘mp_rom_obj_tuple_t’?
 NORETURN void mp_raise_NotImplementedError(mp_rom_error_text_t msg);
                                            ^~~~~~~~~~~~~~~~~~~
                                            mp_rom_obj_tuple_t
<builtin>: recipe for target 'hello-embed.o' failed
make: *** [hello-embed.o] Error 1
make: Leaving directory '/home/workstation/dev/lego/micropython/examples/embedding'
```

### PR DESCRIPTION
**Pull request by hellozyemlya**
Fixes https://github.com/micropython/micropython/issues/5909

### CODE REVIEW

File: examples/embedding/Makefile [hunk 1]
```diff
@@ -6,3 +6,8 @@ hello-embed: hello-embed.o -lmicropython
 
 -lmicropython:
 	$(MAKE) -f $(MPTOP)/examples/embedding/Makefile.upylib MPTOP=$(MPTOP)
+
+clean:
+	rm -rf build
```
Comment (stinos) [inline examples/embedding/Makefile:11]: Normally this should be $(RM) but I'm not sure if it's ok to include mkenv.mk here?

File: examples/embedding/Makefile [hunk 2]
```diff
@@ -6,3 +6,8 @@ hello-embed: hello-embed.o -lmicropython
...
+	rm -f libmicropython.o
```
Comment (stinos) [inline examples/embedding/Makefile:12]: Idem

File: examples/embedding/Makefile [hunk 3]
```diff
@@ -6,3 +6,8 @@ hello-embed: hello-embed.o -lmicropython
...
+	rm -f hello-embed.o
```
Comment (stinos) [inline examples/embedding/Makefile:13]: Note: normally this should be `$(RM)` but I'm not sure if it's ok to include mkenv.mk here?

File: examples/embedding/mpconfigport_minimal.h [hunk 4]
```diff
@@ -132,3 +133,6 @@ typedef long mp_off_t;
 #else
 #include <alloca.h>
 #endif
+
+// Define some macroses
+#define NORETURN
```
Comment (stinos) [inline examples/embedding/mpconfigport_minimal.h:138]: Should have a nawline at the end of the file

File: examples/embedding/mpconfigport_minimal.h [hunk 5]
```diff
@@ -132,3 +133,6 @@ typedef long mp_off_t;
 #else
 #include <alloca.h>
 #endif
+
+// Define some macroses
```
Comment (stinos) [inline examples/embedding/mpconfigport_minimal.h:137]: macroses -> macros

File: examples/embedding/Makefile (hunk 1 above)
Comment (hellozyemlya) [reply examples/embedding/Makefile]: make sense, done

File: examples/embedding/Makefile (hunk 2 above)
Comment (hellozyemlya) [reply examples/embedding/Makefile]: done

File: examples/embedding/mpconfigport_minimal.h (hunk 4 above)
Comment (hellozyemlya) [reply examples/embedding/mpconfigport_minimal.h]: done

File: examples/embedding/Makefile (hunk 3 above)
Comment (hellozyemlya) [reply examples/embedding/Makefile]: I included it, and as far as I see it does not contains something dangerous... But I am not an expert in makefiles as well as in particular micropython case.

File: examples/embedding/mpconfigport_minimal.h (hunk 5 above)
Comment (hellozyemlya) [reply examples/embedding/mpconfigport_minimal.h]: good point, resolved

File: examples/embedding/Makefile [hunk 6]
```diff
@@ -1,8 +1,17 @@
...
 
 -lmicropython:
 	$(MAKE) -f $(MPTOP)/examples/embedding/Makefile.upylib MPTOP=$(MPTOP)
+
+clean:
+	$(MAKE) -f $(MPTOP)/examples/embedding/Makefile.upylib MPTOP=$(MPTOP) clean
+	$(RM) -f hello-embed.o
+	$(RM) -f hello-embed
+
+all: hello-embed
```
Comment (stinos) [inline examples/embedding/Makefile:17]: Missing newline (I'm lookin at this on github and that makes it easy to see that; maybe configure your text editor to automatically add newline at end of file?). For the rest this is all good I think.

Comment (hellozyemlya) [reply examples/embedding/Makefile:17]: Its a strange things in vs code, for boardconfig I manually placed two empty lines to make it looks like one empty line in real file...

**Comment by dpgeorge** [comment]
Thanks for the fix.  I see why MICROPY_ENABLE_DYNRUNTIME is enabled, but it's a bit confusing because this example does not use the dynamic runtime.  To fix this particular item in a different way I made PR #5917 .

File: examples/embedding/Makefile (hunk 6 above)
Comment (hellozyemlya) [reply examples/embedding/Makefile]: fixed

**Comment by hellozyemlya** [comment]
@dpgeorge cool, that will be helpfull. Waiting for your PR to be merged. Will rebase my one later.

**Comment by dpgeorge** [comment]
The issue with missing `mp_rom_error_text_t` typedef is now fixed by 2725a791929f901c692d09f0afe7cc6a700384d5.  So fixing the embedding example should now be much simpler.

**Comment by jimmo** [comment]
hi @hellozyemlya -- in #5964 I've repurposed examples/embedded to be something quite different. As someone who is using this example, I'd be interested to know your thoughts. Do you think I need to add a second example, maybe examples/embedded-unix that retains all the existing functionality based on the Unix port. This wouldn't be too difficult and perhaps would be a good demo of the new approach.

**Comment by arhuaco** [comment]
Hello there. Just run into the issue. In my case an example that gets you started in UNIX is great. Based on the UNIX port would work.

**Comment by dpgeorge** [comment]
A simpler fix was merged in 42035e5eded531d41319ffde999fa5a97478c872, see #6958.

I'll close this PR, but feel free to reopen (or create a new one) if there are additional improvements to the embedding example.

**Pull request closed without merging**